src/librustc_driver/driver.rs

```diff
@@ -804,6 +799,10 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
         })
     })?;
 
+    time(time_passes,
+         "early lint checks",
+         || lint::check_ast_crate(sess, &krate));
```

Comment:
Correct, yes. Presumably we could just empty the "lint buffer" at the end though if need be instead of at the end of the early lint passes.